### PR TITLE
[wasm][debugger] Support multidimensional indexing of object scheme

### LINKED ISSUE #76062: [wasm][debugger] Support multidimensional indexing of `object` scheme
In `MemberReferenceResolver.cs` we are throwing 
```
throw new InvalidOperationException($"Cannot apply indexing with [,] to an object of type '{type}'");
```
Remove it and fix it for e.g. testcase:
```
        [Fact]
        public async Task EvaluateObjectIndexingMultidimensional() => await CheckInspectLocalsAtBreakpointSite(
            "DebuggerTests.EvaluateLocalsWithIndexingTests", "EvaluateLocals", 12, "DebuggerTests.EvaluateLocalsWithIndexingTests.EvaluateLocals",
            "window.setTimeout(function() { invoke_static_method ('[debugger-test] DebuggerTests.EvaluateLocalsWithIndexingTests:EvaluateLocals'); })",
            wait_for_event_fn: async (pause_location) =>
           {
               var id = pause_location["callFrames"][0]["callFrameId"].Value<string>();

               await EvaluateOnCallFrameAndCheck(id,
                   ("f[\'a\', \'&\']", TString("a-&"))
                );
           });
```

```
public class TestEvaluate
{
    public string this[char key1, char key2] => $"{key1}-{key2}";
}
```
and similar. Make sure that not only numeric indexes are supported but other schemes (as above: char) as well.

### PR DESCRIPTION
**Pull request by ilonatommy**
Fixes https://github.com/dotnet/runtime/issues/76062.

Out of scope:
- indexing `valuetype` scheme https://github.com/dotnet/runtime/pull/92637

### CODE REVIEW

File: src/mono/wasm/debugger/tests/debugger-test/debugger-evaluate-test.cs [hunk 1]
```diff
@@ -534,11 +533,14 @@ public class TestEvaluate
             public int this[Indexer indexer] => indexer.index;
             public char this[char[] arr] => arr.Length == 0 ? '0' : arr[0];
 
+            public double this[int key1, double key2] => key1 + key2;
+            public string this[char key1, string key2, string key3] => $"{key1}-{key2}-{key3}";
```
Comment (thaystg) [inline src/mono/wasm/debugger/tests/debugger-test/debugger-evaluate-test.cs:537]: We are testing this one?

Comment (thaystg) [reply src/mono/wasm/debugger/tests/debugger-test/debugger-evaluate-test.cs:537]: I found it! :)

**Review by thaystg — APPROVED** [review]

File: src/mono/wasm/debugger/tests/debugger-test/debugger-evaluate-test.cs (hunk 1 above)
Comment (ilonatommy) [reply src/mono/wasm/debugger/tests/debugger-test/debugger-evaluate-test.cs:537]: Yes, line 760 in tests:
```
"f[aChar, \"&\", longString]", TString("9-&-longString")
```

**Pull request merged**